src/context/AuthContext.tsx

```diff
@@ -45,6 +51,39 @@ export const AuthContextProvider = ({
       })
   }
 
+  function facebookSignIn(auth: Auth) {
+    const facebookProvider = new FacebookAuthProvider()
+    signInWithRedirect(auth, facebookProvider)
+      .then((result) => {
+        return result
+      })
+      .catch((error) => {
+        return error
+      })
+  }
+
+  function twitterSignIn(auth: Auth) {
+    const twitterProvider = new TwitterAuthProvider()
+    signInWithRedirect(auth, twitterProvider)
+      .then((result) => {
+        return result
+      })
+      .catch((error) => {
+        return error
+      })
+  }
+
+  function microsoftSignIn(auth: Auth) {
+    const microsoftProvider = new OAuthProvider('microsoft.com')
+    signInWithRedirect(auth, microsoftProvider)
+      .then((result) => {
+        return result
+      })
+      .catch((error) => {
+        return error
+      })
+  }
+
   function getGoogleResults(auth: Auth) {
     getRedirectResult(auth)
       .then((result) => {
```

Comment:
Is it possible to refactor these functions into one function?

Reply:
Good idea, Just did it 👍🏻